mono/sgen/sgen-gc.h

```diff
@@ -1130,8 +1130,10 @@ typedef enum {
 } SgenAllocatorType;
 
 void sgen_clear_tlabs (void);
-void sgen_update_allocation_count (void);
-guint64 sgen_get_total_allocated_bytes (MonoBoolean precise);
+void sgen_update_allocation_count (void)
+	MONO_PERMIT (need (sten_world_stopped));
+guint64 sgen_get_total_allocated_bytes (MonoBoolean precise)
+	MONO_PERMIT (need (sgen_lock_gc, sgen_stop_world));
```

Comment:
So, I'm not sure if this matters, but sgen_get_total_allcoated_bytes only needs that lock if precise is true. I guess that is ok as a static over approximation though?

Reply:
yea Ward is not at all flow sensitive, so that kind of conditional information is lost on it.